packages/vite/src/node/plugins/worker.ts

```diff
@@ -75,7 +75,7 @@ async function bundleWorkerEntry(
   const { plugins, rollupOptions, format } = config.worker
   const { plugins: resolvedPlugins, config: workerConfig } =
     await plugins(newBundleChain)
-  const workerEnvironment = new BuildEnvironment('client', workerConfig) // TODO: should this be 'worker'?
+  const workerEnvironment = new BuildEnvironment('$client', workerConfig) // TODO: should this be '$worker'?
```

Comment:
I think it is good that later on we have a different key for worker. The current feature is a bit different though, because `$env` represents a single module graph, and `worker` configures any number of workers. @hi-ogawa and @sapphi-red were playing already on using environments to configure workers and if we move there, it would probably be more like (removing `worker`):
```js
export default {
  $processor: workerEnvironment({ ... }),
```
So you would create a new environment of type worker for each of your app workers.

Reply:
> The current feature is a bit different though, because `$env` represents a single module graph, and `worker` configures any number of workers.

I've not tried but I guess `worker` can be a single module graph by using `import`/`importScripts`. If it's fine, I'd like to reserve `$worker`/`$serviceWorker`/`$worklet` for future use.